common/src/ui/RecentDocumentListBox.cpp

```diff
@@ -85,4 +87,22 @@ void RecentDocumentListBox::doubleClicked(const size_t index)
   }
 }
 
+void RecentDocumentListBox::keyPressEvent(QKeyEvent* event)
+{
+  if (
+    (event->key() == Qt::Key_Return || event->key() == Qt::Key_Enter)
+    && (event->modifiers() == Qt::NoModifier || event->modifiers() == Qt::KeypadModifier))
+  {
+
+    if (currentRow() >= 0)
+    {
+      doubleClicked(static_cast<size_t>(currentRow()));
```

Comment:
It feels wrong to call a function named `doubleClicked` from a key press event handler. Please insert a preparation commit before this one where you extract a helper function that takes an index and loads the corresponding recent document -- then call that function from your new key press event handler and from `doubleClicked`.

The helper function needs a descriptive name, maybe `loadRecentDocument`, and take the index. I'm not sure if this name will somehow conflict with the signal of the same name, but it might work. The function should be a private member of `RecentDocumentListBox`, I would put it below the event handlers, but before the signals (in the header file).